### PR TITLE
[feature](insert) Support group commit insert

### PR DESCRIPTION
**Pull request by mymeiyi**
## Proposed changes

Now `INSERT INTO ... VALUES ...` in Doris is a seperate load job, which will begin a transaction and write a rowset, this is costly for FE metadata, and BE io to compact many small rowsets.
This PR is to solve this problem by making multi inserts be loaded in one job.

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/runtime/group_commit_mgr.cpp [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+        return Status::InternalError("instance " + print_id(instance_id) + " not found");
+    }
+    load_instance_info = it->second;
+    return Status::OK();
+}
+
+GroupCommitMgr::GroupCommitMgr(ExecEnv* exec_env) : _exec_env(exec_env) {
+    ThreadPoolBuilder("InsertIntoGroupCommitThreadPool")
+            .set_min_threads(10)
+            .set_max_threads(10)
```
Comment (dataroaring) [inline be/src/runtime/group_commit_mgr.cpp:287]: make it as a config.

**Comment by mymeiyi** [comment]
run buildall

File: be/src/runtime/group_commit_mgr.cpp [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+                      << ", filter rows=" << runtime_state->num_rows_load_filtered()
+                      << ", unselect rows=" << runtime_state->num_rows_load_unselected()
+                      << ", success rows=" << runtime_state->num_rows_load_success();
+        }
+    }
+    int64_t total_rows = 0;
+    int64_t loaded_rows = 0;
+    for (const auto& future_block : future_blocks) {
+        std::unique_lock<doris::Mutex> l(*(future_block->lock));
+        if (!std::get<0>(*(future_block->block_status))) {
```
Comment (dataroaring) [inline be/src/runtime/group_commit_mgr.cpp:427]: future_block->if_handled() is much more readable.

File: be/src/runtime/group_commit_mgr.cpp [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
+        bool eof = false;
+        bool first = true;
+        while (!eof) {
+            // TODO what to do if read one block error
+            RETURN_IF_ERROR(file_scan_node.get_next(runtime_state.get(), _block.get(), &eof));
+            std::shared_ptr<doris::vectorized::FutureBlock> future_block =
+                    std::make_shared<doris::vectorized::FutureBlock>();
+            future_block->swap(*(_block.get()));
+            future_block->set_info(request->base_schema_version(), pipe_id, first, eof);
+            // TODO what to do if add one block error
```
Comment (dataroaring) [inline be/src/runtime/group_commit_mgr.cpp:401]: 3. Put the block into block queue.

File: be/src/runtime/group_commit_mgr.cpp [hunk 4]
```diff
@@ -0,0 +1,480 @@
...
+                      << ", unselect rows=" << runtime_state->num_rows_load_unselected()
+                      << ", success rows=" << runtime_state->num_rows_load_success();
+        }
+    }
+    int64_t total_rows = 0;
+    int64_t loaded_rows = 0;
+    for (const auto& future_block : future_blocks) {
+        std::unique_lock<doris::Mutex> l(*(future_block->lock));
+        if (!std::get<0>(*(future_block->block_status))) {
+            future_block->cv->wait(l);
```
Comment (dataroaring) [inline be/src/runtime/group_commit_mgr.cpp:428]: 4 wait to wal

**Review by dataroaring — CHANGES_REQUESTED** [review]

File: be/src/runtime/group_commit_mgr.h [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    int64_t txn_id;
+    int64_t schema_version;
+    bool need_commit = false;
+
+private:
+    std::chrono::steady_clock::time_point _start_time;
+
+    std::shared_ptr<doris::Mutex> _mutex;
+    std::shared_ptr<doris::ConditionVariable> _cv;
+    std::set<UniqueId> _block_unique_ids;
```
Comment (dataroaring) [inline be/src/runtime/group_commit_mgr.h:68]: rename it to _load_unique_ids.

**Review by dataroaring — CHANGES_REQUESTED** [review]

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.51 seconds
 stream load tsv:          540 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17162150703 Bytes

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.04 seconds
 stream load tsv:          542 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162136301 Bytes

**Comment by mymeiyi** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.56 seconds
 stream load tsv:          552 seconds loaded 74807831229 Bytes, about 129 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          31.3 seconds inserted 10000000 Rows, about 319K ops/s
 storage size: 17162165047 Bytes

**Comment by dataroaring** [comment]
run buildall

**Comment by mymeiyi** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by Yukang-Lian — APPROVED** [review]

**Pull request merged**